packages/components/src/components/tablist/tablist.component.ts

```diff
@@ -0,0 +1,430 @@
+ * - If a `tab` element has a popup menu, it needs to have the property `aria-haspopup` set to either `menu` or `true`.
+ *
+ * @tagname mdc-tablist
+ *
+ * @dependency mdc-tab
+ * @dependency mdc-button
+ *
+ * @event change - (React: onChange) This event is dispatched when the tab is selected.
+ *
+ *
+ * Event that fires when user changes the active tab.
+ *     The function sent as the argument will receive the fired event
+ *      and the new tab id can be fetched from event.detail.tabId.
+ *
+ *    `(event: CustomEvent) => handleTabChange(event.detail.tabId);`
+ *
+ * @slot Default slot for mdc-tab elements.
+ *
+ * @cssproperty --mdc-tablist-gap - Gap between tabs
+ * @cssproperty --mdc-tablist-width - Width of the tablist
+ * @cssproperty --mdc-container-button-padding - Padding for when an arrow button is visible
+ */
+class TabList extends Component {
+  /**
+   * ID of the active tab, defaults to the first tab if not provided
+   */
+  @property({ type: String, attribute: 'active-tabid', reflect: true })
+  activeTabId?: string;
+
+  @query('.container')
+  private tabsContainer?: HTMLDivElement;
+
+  @queryAssignedElements({ selector: 'mdc-tab' })
+  private tabs?: Tab[];
+
+  /**
+   * Instead of using 'left' and 'right' for notation of the arrow buttons, we use 'forward' and 'backward'.
+   * Since the elements of the tablist gets flipped in RTL layouts, the arrow buttons also flip.
+   * For both RTL and LTR layouts, the forward arrow button moved the tabs in the forward direction.
+   * i.e. the forward arrow button moves the tabs to the right in LTR layouts and to the left in RTL layouts.
+   */
+
+  /**
+   * @internal
+   */
+  @state() private showForwardArrowButton: boolean = false;
+
+  /**
+   * @internal
+   */
+  @state() private showBackwardArrowButton: boolean = false;
+
+  private get isRtl(): boolean {
+    return (
+      document.querySelector('html')?.getAttribute('dir') === 'rtl' || window.getComputedStyle(this).direction === 'rtl'
+    );
+  }
+
+  constructor() {
+    super();
+
+    this.addEventListener('keydown', this.handleKeydown);
+    // Reason for assertion below: https://github.com/microsoft/TypeScript/issues/28357
+    this.addEventListener('activechange', this.handleNestedTabActiveChange as EventListener);
+  }
+
+  override connectedCallback() {
+    super.connectedCallback();
+    this.role = 'tablist';
+  }
+
+  /**
+   * Set the initial active tab after firstUpdated,
+   * since this.tabs (=slot) are only available after
+   * the first update
+   *
+   * Also set up the event listeners.
+   */
+  protected override firstUpdated(): void {
+    /**
+     * If there are no tabs, skip the initialization of the active tab and the event listeners.
+     * Then throw an error.
+     */
+    if (!this.tabs) { return; }
+    if (Array.isArray(this.tabs) && this.tabs.length === 0) {
+      throw new Error('The tablist component must have at least one child tab');
```

Comment:
```suggestion
     if(this.onerror){
        this.onerror('The tablist component must have at least one child tab');
     }
```
We cannot throw errors from a web component.

Reply:
Replaced